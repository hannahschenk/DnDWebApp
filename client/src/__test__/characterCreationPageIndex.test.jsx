import React, { useState as useStateMock } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Index from "../pages/characterCreationPage/index";


// Upon test, character is undefined thus the destructuring of undefined is causing an error when testing.
//For now comment out just to verify that form is rendering properly.
describe("The characterCreationPage Component...", () => {
    test("will render without crashing", () => {
        const { baseElement } = render(<Index />);
        expect(baseElement).toBeDefined();
    })
 
    test("will match snapshot",()=>{
        // renderer will give you a pure JS object you can turn in to JSON of the page the has rendered.
        const tree = renderer.create(<Index />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})