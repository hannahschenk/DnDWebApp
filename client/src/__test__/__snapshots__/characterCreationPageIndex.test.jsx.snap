// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The characterCreationPage Component... will match snapshot 1`] = `
<section>
  This is the guided character Creation process. Please fill out each section before clicking next.
  <button
    onClick={[Function]}
  >
    Start Character Creation
  </button>
</section>
`;
